### PR TITLE
Javacore5

### PR DESCRIPTION
**Pull request by k-kudryavtseva**
- Separated entities into different files;
- added exceptions;
- added hierarchy of engine and wheel;
- modified calculation of current speed;

### CODE REVIEW

File: CountryRoad.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package JavaAutomationLab;
+
+public class CountryRoad extends RouteSection {
```
Comment (shukal94) [inline CountryRoad.java]: let's remove that on future task - no need to make hierarchy here

File: PointLocation.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package JavaAutomationLab;
+
+public class PointLocation {
+    private int x = 0;
```
Comment (shukal94) [inline PointLocation.java:4]: how do you going to work with that vars?

File: VehicleFactory.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package JavaAutomationLab;
+
+import java.lang.reflect.InvocationTargetException;
+
+public class VehicleFactory {
```
Comment (shukal94) [inline VehicleFactory.java]: let's make it abstract

File: PointLocation.java (hunk 2 above)
Comment (k-kudryavtseva) [reply PointLocation.java:4]: can't come up with an idea why we need 2D points in the task, seems like 1D is enough to track progress in the race. That's why this class is not used for now.

File: CountryRoad.java (hunk 1 above)
Comment (k-kudryavtseva) [reply CountryRoad.java]: > let's remove that on future task - no need to make hierarchy here

Modified

File: VehicleFactory.java (hunk 3 above)
Comment (k-kudryavtseva) [reply VehicleFactory.java]: > let's make it abstract

Modified

**Pull request merged**